src/nvim/shada.c

```diff
@@ -2402,6 +2413,46 @@ static inline ShaDaWriteResult shada_read_when_writing(
   return ret;
 }
 
+/// Initializes registers for writing to the ShaDa file
+///
+/// @param[in]  wms  The WriteMergerState used when writing.
+/// @param[in]  max_reg_lines The maximum number of register lines.
+static void shada_initialize_registers(WriteMergerState *const wms,
+                                       int max_reg_lines) {
```

Comment:
Should be

```C
static inline void shada_initialize_registers(WriteMergerState *const wms
                                                  const int max_reg_lines)
  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_ALWAYS_INLINE
{
```

: missing inlining (optional, but better have it), NONNULL_ALL attribute and `{` must be on its own line.